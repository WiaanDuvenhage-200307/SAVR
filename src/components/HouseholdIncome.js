import React from 'react';
import Button from './UI/Button';
import { IconButton } from '@mui/material';
import DeleteIcon from '@mui/icons-material/Delete'
import { useState } from 'react';
import Input from './UI/Input';

const HouseholdIncome = () => {

    const [formValues, setFormValues] = useState([0]);

    const [status, setStatus] = useState("normal");

    const handleInputChange = (newNumber, index) => {
        let changingFormValue = formValues;
        changingFormValue[index] = newNumber;

        console.log(changingFormValue);
        setFormValues(changingFormValue);
    }
    
    const addNewNumber  = () =>{
        setFormValues([...formValues, 0])
    }

    
    const removeNumberFromArray = (index) => {
        var changingFormValue = formValues;
        setFormValues(changingFormValue.splice(index, 1));
        console.log("clicked");
    }

    return (
        <div className='house-income'>
            <h1 className='house-income__heading'>HOUSEHOLD INCOME</h1>
            <div className='house-income__members'>
                {formValues.map((number, index) => (
                    <div key={index}>
                        <Input className='house-income-input' onChange={(e) => handleInputChange(e.target.value)} ariaLabel='names' type='text' placeholder='Full Name'/>
                        <label htmlFor='amount'>$</label>
                        <Input className='house-income-input' onChange={(e) => handleInputChange(+e.target.value)} name='amount' ariaLabel='income' type='text' placeholder='Amount'/>
                        <IconButton aria-label="delete" onClick={removeNumberFromArray}><DeleteIcon/></IconButton>
                    </div>
                ))}
                
            </div>
            <Button function={()=> addNewNumber(true)}  name="Add Member" className="button" onMouseEnter={() => setStatus("hovered")} onMouseLeave={() => setStatus("normal")} role="Button" btnName="addMemberBtn"/>
        </div>
    );
};

export default HouseholdIncome;