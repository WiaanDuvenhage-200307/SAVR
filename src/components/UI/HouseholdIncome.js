import React from 'react';
import Button from './Button';
import { IconButton } from '@mui/material';
import DeleteIcon from '@mui/icons-material/Delete'

const HouseholdIncome = () => {
    return (
        <div className='house-income'>
            <h1 className='house-income__heading'>HOUSEHOLD INCOME</h1>
            <div className='house-income__members'>
                <h2>John Doe</h2>
                <div className='house-income_delete'>
                    <h2 className='house-income__money'>$ 3900</h2>
                    <IconButton aria-label="delete"><DeleteIcon/></IconButton>
                </div>
            </div>
            <Button name="Add +" className="button"/>
        </div>
    );
};

export default HouseholdIncome;