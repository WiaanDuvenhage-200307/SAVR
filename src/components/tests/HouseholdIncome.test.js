import HouseholdIncome from "../UI/HouseholdIncome";
import { render, screen } from "@testing-library/react";

describe("Testing our form interaction...", () => {

    test("checking to see if our inputs are empty...", () => {
        render(<HouseholdIncome/>) //rendering in the HouseHoldIncome Component

        const fullNameInput = screen.getByLabelText(/names/i); //targeting the 'names' input
        const amountInput = screen.getByLabelText(/income/i); //targeting the 'income' input

        expect(fullNameInput.value).toBe(""); //checking to see if the input is empty
        expect(amountInput.value).toBe(""); //checking to see if the input is empty
    })
})
