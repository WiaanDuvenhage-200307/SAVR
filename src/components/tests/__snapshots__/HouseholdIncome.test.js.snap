// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing our form interaction... See if button changes color.. 1`] = `
<button
  className="button"
  name="Add Member"
>
  Add Member
</button>
`;
